keras_cv/models/resnet_v2.py

```diff
@@ -99,210 +101,312 @@
+            "relu", name=self.name + "_use_preactivation_relu"
+        )
         if conv_shortcut:
-            shortcut = layers.Conv2D(
+            self.shortcut = layers.Conv2D(
                 4 * filters,
                 1,
-                strides=s,
-                name=name + "_0_conv",
-            )(use_preactivation)
+                strides=self.stride2,
+                name=self.name + "_0_conv",
+            )
         else:
-            shortcut = (
-                layers.MaxPooling2D(1, strides=stride, name=name + "_0_max_pooling")(x)
-                if s > 1
-                else x
+            self.shortcut = (
+                layers.MaxPooling2D(
+                    1, strides=stride, name=self.name + "_0_max_pooling"
+                )
+                if self.stride2 > 1
+                else None
             )
-
-        x = layers.Conv2D(filters, 1, strides=1, use_bias=False, name=name + "_1_conv")(
-            use_preactivation
-        )
-        x = layers.BatchNormalization(
-            axis=BN_AXIS, epsilon=1.001e-5, name=name + "_1_bn"
-        )(x)
-        x = layers.Activation("relu", name=name + "_1_relu")(x)
-
-        x = layers.Conv2D(
-            filters,
-            kernel_size,
-            strides=s,
+        self.conv1 = layers.Conv2D(
+            self.filters,
+            kernel_size=1,
+            strides=1,
             use_bias=False,
+            name=self.name + "_1_conv",
+        )
+        self.bn1 = layers.BatchNormalization(
+            axis=BN_AXIS, epsilon=BN_EPSILON, name=self.name + "_1_bn"
+        )
+        self.activation1 = layers.Activation("relu", name=self.name + "_1_relu")
+        self.conv2 = layers.Conv2D(
+            self.filters,
+            self.kernel_size,
+            strides=self.stride2,
             padding="same",
-            dilation_rate=dilation,
-            name=name + "_2_conv",
-        )(x)
-        x = layers.BatchNormalization(
-            axis=BN_AXIS, epsilon=1.001e-5, name=name + "_2_bn"
-        )(x)
-        x = layers.Activation("relu", name=name + "_2_relu")(x)
-
-        x = layers.Conv2D(4 * filters, 1, name=name + "_3_conv")(x)
-        x = layers.Add(name=name + "_out")([shortcut, x])
-        return x
-
-    return apply
+            dilation_rate=self.dilation,
+            use_bias=False,
+            name=self.name + "_2_conv",
+        )
+        self.bn2 = layers.BatchNormalization(
+            axis=BN_AXIS, epsilon=BN_EPSILON, name=self.name + "_2_bn"
+        )
+        self.activation2 = layers.Activation("relu", name=self.name + "_2_relu")
+        self.conv3 = layers.Conv2D(
+            4 * filters, kernel_size=1, name=self.name + "_3_conv"
+        )
+        self.add = layers.Add(name=self.name + "_out")
+
+    def call(self, inputs):
+        x = self.bn_preactivation(inputs)
+        x = self.relu_preactivation(x)
+        shortcut = self.shortcut(x) if self.shortcut else x
```

Comment:
After porting the weights and finding numerical inconsistencies, I discovered that this line is incorrect.

It should be:

```
if self.conv_shortcut:
          shortcut = self.shortcut(x)
        elif self.shortcut:
          shortcut = self.shortcut(inputs)
        else:
          shortcut = inputs
```

Reply:
(same for BasicBlock btw)